packages/core-v2-graphql/src/resolvers/index.ts

```diff
@@ -1,69 +1,89 @@
-import { CoreV2Context } from "src/context";
-import { MultiTerm, Resolvers } from "src/generated";
-import {
-  PoolModel,
-  TokenModel,
-  TermModel,
-  YieldSourceModel,
-  MultiPoolModel,
-  MultiTermModel,
-} from "src/models";
-
-export const resolvers: Resolvers<CoreV2Context> = {
+import { MultiPool, MultiTerm, Token } from "@elementfi/core-v2-sdk";
+import { ElementGraphQLContext } from "src/context";
+import * as types from "src/generated";
+
+export const resolvers: types.Resolvers<ElementGraphQLContext> = {
   Query: {
-    multiTerm: (_, { address, yieldSource: yieldSourceName }, context) => {
-      let multiTerm: MultiTerm | null = null;
+    multiTerm: (_, { address, yieldSource }, { elementContext }) => {
       if (address) {
-        multiTerm = MultiTermModel.getByAddress({ address, context });
-      } else if (yieldSourceName) {
-        const yieldSource = YieldSourceModel.getByName({
-          name: yieldSourceName,
-          context,
-        });
-        multiTerm = MultiTermModel.getByYieldSource({
-          yieldSource,
-          context,
-        });
+        return new MultiTerm(address, elementContext);
+      } else if (yieldSource) {
+        // TODO: How will yield source names be mapped to multiterms?
       }
-      return multiTerm || null;
+      return null;
     },
 
-    term: (_, { multiTerm: multiTermAddress, maturity }, context) => {
-      const multiTerm = MultiTermModel.getByAddress({
-        address: multiTermAddress,
-        context,
-      });
-      const term = TermModel.getByMaturity({ maturity, multiTerm, context });
-      return term || null;
+    term: async (
+      _,
+      { multiTerm: multiTermAddress, maturity },
+      { elementContext },
+    ) => {
+      const term = await new MultiTerm(
+        multiTermAddress,
+        elementContext,
+      ).getTerm(maturity / 1e3);
+      if (term) {
+        const { id: idNumber, maturityDate, multiTerm } = term;
+        const maturity = maturityDate.getTime();
+        const id = idNumber.toString();
+        return {
+          id,
+          multiTerm,
+          maturity,
+          principalToken: {
+            id,
+            maturity,
+          },
+        };
+      }
+      return null;
```

Comment:
we can just create a new term with a constructor instead of going through the getTerm method. Currently it does not check if a term `exists`. Honestly I think you should be able to create a term model even if there is no liquidity at that expiry. For example what if we want to use the model to lock tokens and you are the seeder?

Reply:
But the new `Term` constructor still requires a `MultiTerm` instance as its 3rd arg. So, `MultiTerm.getTerm` is the same as `new Term(..., ..., MultiTerm)`. That's a good point about being able to create an model instance for a term that doesn't exists though.